clang/include/clang/CConv/TypeVariableAnalysis.h

```diff
@@ -0,0 +1,66 @@
+//=--TypeVariableAnalysis.h---------------------------------------*- C++-*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_CLANG_LIB_CCONV_TYPEVARIABLEANALYSIS_H
+#define LLVM_CLANG_LIB_CCONV_TYPEVARIABLEANALYSIS_H
+
+#include <set>
+#include "ConstraintVariables.h"
+
+class TypeVariableEntry {
+public:
+  // Note: does not initialize TyVarType!
+  TypeVariableEntry() :
+      IsConsistent(false) {}
+  TypeVariableEntry(QualType Ty, std::set<ConstraintVariable *> &CVs) {
+    if (isTypeAnonymous(Ty->getPointeeType())) {
+      IsConsistent = false;
+    } else {
+      IsConsistent = true;
+      TyVarType = Ty;
+      ArgConsVars = CVs;
+    }
+  }
+
+  bool getIsConsistent();
+  QualType getType();
+  std::set<ConstraintVariable *> &getConstraintVariables();
+  ConstraintVariable *getTypeParamConsVar();
+
+  void insertConstraintVariables(std::set<ConstraintVariable *> &CVs);
+  void setTypeParamConsVar(ConstraintVariable *CV);
+  void updateEntry(QualType Ty, std::set<ConstraintVariable *> &CVs);
+
+private:
+  // Is this type variable used consistently. True when all uses have the same
```

Comment:
Instead of saying "used" maybe write "instantiated" ? I think that's the standard technical term for what you are getting at.